controllers/memberoperatorconfig/memberoperatorconfig_controller.go

```diff
@@ -26,7 +27,23 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 	}
 
 	// Watch for changes to primary resource MemberOperatorConfig
-	return c.Watch(&source.Kind{Type: &toolchainv1alpha1.MemberOperatorConfig{}}, &handler.EnqueueRequestForObject{}, &predicate.GenerationChangedPredicate{})
+	if err = c.Watch(
+		&source.Kind{Type: &toolchainv1alpha1.MemberOperatorConfig{}},
+		&handler.EnqueueRequestForObject{},
+		&predicate.GenerationChangedPredicate{},
+	); err != nil {
+		return err
+	}
+
+	// Watch for changes to secrets that should trigger an update of the config cache
+	// +kubebuilder:rbac:groups="",resources=secrets,verbs=get;watch;list
+	return c.Watch(
+		&source.Kind{Type: &corev1.Secret{}},
+		&handler.EnqueueRequestsFromMapFunc{
+			ToRequests: SecretToMemberOperatorConfigMapper{client: mgr.GetClient()},
```

Comment:
And here too. You don't need a client here.

Reply:
Removed it with the previous commit: cc1eb37ab554057e49588fa4b56fc1bbcda08a8f